tests/base/src/test/java/org/keycloak/tests/admin/SMTPConnectionTest.java

```diff
@@ -15,53 +15,65 @@
  * limitations under the License.
  */
 
-package org.keycloak.testsuite.admin;
+package org.keycloak.tests.admin;
 
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
+import org.keycloak.admin.client.Keycloak;
 import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.models.AdminRoles;
+import org.keycloak.models.Constants;
 import org.keycloak.representations.idm.RealmRepresentation;
-import org.keycloak.representations.idm.UserRepresentation;
-import org.keycloak.testsuite.AbstractKeycloakTest;
-import org.keycloak.testsuite.util.GreenMailRule;
-import org.keycloak.testsuite.util.UserBuilder;
 
 import jakarta.mail.internet.MimeMessage;
 import jakarta.ws.rs.core.Response;
+import org.keycloak.testframework.annotations.InjectAdminClient;
+import org.keycloak.testframework.annotations.InjectRealm;
+import org.keycloak.testframework.annotations.KeycloakIntegrationTest;
+import org.keycloak.testframework.mail.MailServer;
+import org.keycloak.testframework.mail.annotations.InjectMailServer;
+import org.keycloak.testframework.realm.ManagedRealm;
+import org.keycloak.testframework.realm.RealmConfig;
+import org.keycloak.testframework.realm.RealmConfigBuilder;
+
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
 import static org.keycloak.representations.idm.ComponentRepresentation.SECRET_VALUE;
 
 /**
  * @author <a href="mailto:bruno@abstractj.org">Bruno Oliveira</a>
  */
-public class SMTPConnectionTest extends AbstractKeycloakTest {
+@KeycloakIntegrationTest
+public class SMTPConnectionTest {
 
-    public final String SMTP_PASSWORD = setSmtpPassword();
+    @InjectRealm(config = SMTPRealmWithClientAndUser.class)
+    private ManagedRealm managedRealm;
 
-    @Rule
-    public GreenMailRule greenMailRule = new GreenMailRule();
-    private RealmResource realm;
+    @InjectAdminClient(mode = InjectAdminClient.Mode.MANAGED_REALM, client = "myclient", user = "myadmin")
+    private Keycloak adminClient;
```

Comment:
No need for an admin client here. It's only used to get the `RealmResource`, which can be done with `managedRealm.admin()`

Reply:
Already discussed -> this should remain since it goes about the admin client created for default realm.

Reply:
To explain a bit more this is needed as the test email endpoint relies on sending a test email to the authenticated user invoking the endpoint; so this needs an actual user with an email configured and not the default admin client which is just a service account.